Project 4/SPARQL-JV.md

```diff
@@ -0,0 +1,389 @@
+- Give the name of the institution and number of students of a post-secondary educational institution located in California with the highest enrollment of students (undergrad and grad together).
+```
+PREFIX onto-ed: <https://www.nj.gov/admin/oit/ontology/>
+
+SELECT ?institution (MAX(?totalEnrollment) AS ?topEnrollment)
+WHERE {
+  ?institution a onto-ed:EducationalInstitution ;
+               onto-ed:type "PostsecondaryEducation" ;
+               onto-ed:located_in_state "California" ;
+               onto-ed:enrollment_number ?postgradEnrollment ;
+               onto-ed:enrollment_number ?undergradEnrollment ;
+               BIND (?postgradEnrollment + ?undergradEnrollment AS ?totalEnrollment) .
+}
+```
+**Problem 5-3.**
+Register a block of unused VINs.
+
+Take fields from multiple tables to generate a unique VIN. The third field is a check-digit, which in this case will be "9". Your manufacturing plant will only be assigning 1000. Lastly, verify that the output is exactly 17 characters long.
+
+The order of VIN construction: 
+
+WMI (ontomobile:worldManufacturerIdentifier)
+Vehicle attributes (ontomobile
+Check digit
+Model year
+Plant code
+Sequential number
+
+
+**Problem 5-4.**
+commerce: lettuce that was transported through AR or NE at risk for E Coli.
+construct: "recall" status for all such lettuce.
+- borked: to be filled in
+```
+borked
+```
+
+
+**Problem 5-5.**
+medical: person has genotype (string)
+         person has genotype (string)
+    construct: person has_predisposition testicularcancer
+- borked: to be filled in
+```
+borked
+``` 
+    
+
+**Problem 4-1.**
+- Around the world there are large buildings that bear the same shape as the famous Egyptian pyramids. These are called ziggurats. However, Egyptians were not the first to build this kind of structure. Which structures have the greatest estimated age?
+- Display the name of the oldest 25 ziggurats, current country location, name of associated culture when it was formed, and estimated age. Order by estimated age, descending (or date of completion ascending).
+```
+PREFIX anthro: <http://anthropologywiki.org/schema/ontology/>
+
+SELECT ?zigguratName ?countryName ?cultureName ?estimatedCompletionDate
+WHERE {
+  ?ziggurat a anthro:Building ;
+            rdfs:type anthro:Ziggurat ;
+            anthro:location_in ?country ;
+            anthro:cultural_originator ?culture ;
+            anthro:formation_datecompleted ?estimatedCompletionDate .
+  ?country rdfs:label ?countryName .
+  ?culture rdfs:label ?cultureName .
+  FILTER(lang(?cultureName) = 'en')
+}
+ORDER BY ASC(?estimatedCompletionDate)
+LIMIT 25
+```
+
+**Problem 4-2.**
+- Find all public elections for president held in sovereign states that were formerly members of the USSR and not currently dissolved. Display the country and the winner as well.
+- Note: be sensitive to the open world assumption.
+```
```

Comment:
This is also interesting for the use of the status dissolved. Again, not sure whether the complexity is proper for a level 4.